### PR TITLE
add rmsprop optimizer for python

### PR DESCRIPTION
**Pull request by Kublai-Jing**
Given that many papers have reported RMSProp works the best for RNN/LSTM compared to SGD+momentum, Adagrad, AdaDelta,.... , this might be beneficial for people who work on LSTM models. There are >1 versions of RMSProp. The one I follow is in http://arxiv.org/pdf/1308.0850v5.pdf by Alex Graves.

### CODE REVIEW

**Comment by piiswrong** [comment]
Looks like you changed the default optimizer in LSTM. Could you post the results before/after this change?

**Comment by Kublai-Jing** [comment]
hmm.. sorry I didn't intend to commit those file. those are just my own test

File: python/mxnet/optimizer.py [hunk 1]
```diff
@@ -300,7 +300,94 @@ def update(self, index, weight, grad, state):
...
+        self.time_first_index = None
+    def create_state(self, index, weight):
+        """Create additional optimizer state: mean, variance
+        Parameters
+        ----------
+        weight : NDArray
+            The weight data
+
+        """
+        self.time_first_index = None  # time is incremented only on the first index
```
Comment (piiswrong) [inline python/mxnet/optimizer.py]: what is this for?

**Comment by piiswrong** [comment]
I see. I would still be nice if you can post your results with rms optimizer

**Comment by antinucleon** [comment]
I will merge first then fix the lint problem.

**Pull request merged**